File: traittable/fib/fib.ml
let rec fib n =
  if n < 2 then
    1
  else
    fib(n-2)+fib(n-1)

class f(n) = object
  val _n = n
  method fib =
    if _n < 2 then
      1
    else
       ((new f(_n-2))#fib) + ((new f(_n-1))#fib)
end

let tim ():int = int_of_float(Sys.time() *. 1000.0)

let _ =
  let start = tim() in
  Printf.printf "%d\n" (fib(40));
  Printf.printf "%dms\n" ((tim())-start);
  let start = tim() in
  Printf.printf "%d\n" ((new f(40))#fib);
  Printf.printf "%dms\n" ((tim())-start);

File: traittable/fib/fib.php
<?php

function fib($n) {
  if ($n < 2) return 1;
  return fib($n-2)+fib($n-1);
}

class Fib {
  function __construct($n) {
    $this->n = $n;
  }
 
  function fib() {
    if ($this->n < 2) return 1;
    return new Fib($this->n-2).fib() + new Fib($this->n-1).fib();
  }
}

function tim() {
	return (int)(microtime(true) * 1000);
}

$start = tim();
printf("%d\n", fib(40));
printf("%dms\n",tim()-$start);

$start = tim();
printf("%d\n", new Fib(40).fib());
printf("%dms\n",tim()-$start);
